### ISSUE TITLE
Laser Linewidth

### ISSUE THREAD

**Issue by pathfinder49** [OxfordIonTrapGroup/ion_phys#17]
Currently the package assumes lasers to have infitessimal linewidth. This seems reasonable, given the current package feature-set.

When moving to more advanced (optical bloch equation) simulations, laser linewidth may become relevant.
* Lasercooling, readout, etc.: laser linewidth should be negligible when driving E1 transitions.
* DarkResonaces: laser linewidth may become important with lower laser intensities. However, I expect our cooling scheme to be outside this regime due to large laser powers.

**Comment by RHanley1** [comment]
I think with our expected laser linewidths of ~300kHz, I wouldn't worry about it for now. Just focus getting everything else working. Laser linewidths can be added rather simply later on in a phenomenological way whereby an extra dephasing is introduced on the coherence's between connected levels.

**Comment by hartytp** [comment]
Agreed: don’t bother with them for now as we can add a dephasing  term later. 

FWIW the effective laser linewidth is broadened by the ions motion. Eg around a dark state the ion gets an oscillating Doppler shift that introduces scattering. It’s even more complicated when the d states are introduced as the repumping time can be long. So when the ion reaches the d states the scattering ceases for a while and then re begins with an essentially random phase. It’s messy (see the paper we wrote on this).

In practice we’ve tended to add an empirical dephasing factor that we tweak to make the sims match the data.
